src/tests/system/tests/test_local_overrides.py

```diff
@@ -0,0 +1,263 @@
+        uid=999011, gid=999011, home="/home/user1", gecos="user", shell="/bin/bash", password="Secret123")
+    client.sssd.start()
+    assert client.ssh("user1", "Secret123")
+    sss_override = client.override.user("user1")
+
+    sss_override.add(name="o-user1")
+    client.sssd.restart()
+    assert client.ssh("user1", "Secret123")
+    assert client.ssh("o-user1", "Secret123")
+    assert client.ssh(f"user1@{client.sssd.domain}", "Secret123")
+    assert client.ssh(f"o-user1@{client.sssd.domain}", "Secret123")
+
+    assert 999999 != client.tools.getent.passwd("user1").uid
+    assert 888888 != client.tools.getent.passwd("user1").gid
+    sss_override.modify(name="o-user1", uid=999999, gid=888888, home="/home/o-user1")
+    assert 999999 == client.tools.getent.passwd("user1").uid
+    assert 888888 == client.tools.getent.passwd("user1").gid
+    assert "/home/o-user1" == client.tools.getent.passwd("user1").home
+
+    sss_override.delete()
+    assert sss_override.get() == {}:
+        1. Create group "group1" with posix attributes defined
+        2. Start SSSD
+    :steps:
+        1. Override "group1" to "o-group1"
+        2. Search for group override
+        3. Override posix attributes for "o-group1"
+        4. Search for group override
+        5. Delete override
+    :expectedresults:
+        1. Group local override is created
+        2. Local override for "o-group1" found
+        3. Local override posix attribute updated
+        4. Local override "o-group1" found
+        5. Local override is deleted
+    :customerscenario: False
+    """
+    provider.group("group1").add(gid=999999)
+    client.sssd.start()
+
+    client.local_override.group("group1").add(name="o-group1")
+    client.sssd.restart()
+
+    assert client.tools.getent.group("group1")
+    assert client.tools.getent.group("o-group1")
+    assert client.tools.getent.group("group1").gid != 888888
+    client.local_override.group("group1").modify(gid=888888
+)
+    assert client.tools.getent.group("group1").gid == 888888
+
+    client.local_override.group("group1").delete()
+    assert client.local_override.group("group1").get() == {}
+
+
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.AD)
+def test_local_overrides__root_user(client: Client, provider: GenericProvider):
+    """
+    :title: Local override root user
+    :setup:
+        1. Start SSSD
+    :steps:
+        1. Create local override "root" for the user1 user
+    :expectedresults:
+        1. Local override is created but the root user contents are not modified
+    :customerscenario: False
+    """
+    provider.user("user1").add(
+        uid=999011, gid=999011, home="/home/user1", gecos="user", shell="/bin/bash", password="Secret123")
+    client.sssd.restart()
+
+    client.local_override.user("user1").add(name="root", uid=999022, gid=999022)
+    assert client.tools.getent.passwd("root").uid == 0
+    assert client.tools.getent.passwd("root").gid == 0
```

Comment:
Should you delete the local override created here?

Reply:
Yup